Sources/EpoxyCore/SwiftUI/LayoutUtilities/SizingContainer.swift

```diff
@@ -0,0 +1,77 @@
+// Created by Bryn Bodayle on 1/24/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+import SwiftUI
+
+// MARK: - SizingContainer
+
+/// A container which reads the proposed SwiftUI layout and passes it a `SizingContext`. This stores
+/// ideal size as state in order to configure the view layout as the ideal size is updated via the
+/// `SizingContext`.
+public struct SizingContainer<Content: View>: View {
+
+  // MARK: Lifecycle
+
+  /// Constructs a `SizingContainer` view
+  /// - Parameters:
+  ///   - estimatedSize: An estimated size used as a placeholder ideal size until view measurement
+  ///     occurs. Pass `nil` for this parameter if this container is only used for reading the
+  ///     proposed size and an ideal size will never be provided.
+  ///   - content: The view content to wrap and provide a `SizingContext` to.
+  public init(
+    estimatedSize: CGSize? = CGSize(width: 375, height: 150),
```

Comment:
Do you think it makes sense to break this out into separate width and height `CGFloat?`s? E.g. So that we can support an estimated height but no estimated width?